marker_adapter/src/context/map.rs

```diff
@@ -0,0 +1,76 @@
+use marker_api::{
+    ast::item::{EnumVariant, Field},
+    context::{AstMap, AstMapCallbacks},
+    ffi,
+    lint::Level,
+    prelude::*,
+};
+
+#[repr(C)]
+pub struct AstMapWrapper<'ast> {
+    pub driver: &'ast dyn AstMapDriver<'ast>,
```

Comment:
What's the reason to make this pub as well as in `MarkerContextWrapper`?

Reply:
I think it just made it a bit easier than adding a constructor or builder. There is no specific reasoning besides that.

Reply:
Well, if we consider `marker_adapter` to be entirely private to marker, then it's fine, but if there is a chance this may become stable for external users, I'd refrain from exposing a public field

Reply:
`marker_adapter` is only targeted towards drivers, but if this is a concern, it's easy to just add a constructor. I'll do that in the next commit.